awx/main/utils/ha.py

```diff
@@ -10,26 +10,27 @@
 from awx.main.models import Instance
 
 
-def _add_remove_celery_worker_queues(app, instance, worker_queues, worker_name):
+def _add_remove_celery_worker_queues(app, controlled_instances, worker_queues, worker_name):
     removed_queues = []
     added_queues = []
-    ig_names = set(instance.rampart_groups.values_list('name', flat=True))
-    ig_names.add("tower_instance_router")
+    ig_names = set(['tower_instance_router'])
+    hostnames = set([instance.hostname for instance in controlled_instances])
+    for instance in controlled_instances:
+        ig_names.update(instance.rampart_groups.values_list('name', flat=True))
     worker_queue_names = set([q['name'] for q in worker_queues])
 
-
     # Remove queues that aren't in the instance group
     for queue in worker_queues:
         if queue['name'] in settings.AWX_CELERY_QUEUES_STATIC or \
                 queue['alias'] in settings.AWX_CELERY_QUEUES_STATIC:
             continue
 
-        if queue['name'] not in ig_names | set([instance.hostname]) or not instance.enabled:
+        if queue['name'] not in ig_names | hostnames or not instance.enabled:
```

Comment:
I worry about the 2-phase enable/disable of a node. Phase 1 the user marks the node enabled/disabled. Phase 2, the heartbeat will set the capacity to 0 and actually take the node offline. Stopping the node from listening to the queue here based on Phase 1 eliminates the draining feature that we claim to implement.